tests/Database/DatabaseEloquentBuilderTest.php

```diff
@@ -1460,6 +1460,22 @@ public function testHasWithConstraintsWithOrWhereAndHavingInSubquery()
         $this->assertEquals(['larry', '90210', '90220', 'fooside dr', 29], $builder->getBindings());
     }
 
+    public function testHasWithConstraintsWithOrWhereAndSubqueryInFromInSubquery()
```

Comment:
You've only added a unit test to check the non-expression implementation - which worked before your change. But you also need to add a unit test to check the usage with expressions. Then the former issue would have been detected.

Reply:
Thank you for the feedback. I see I misunderstood the location in my local code that was triggering the issue. I've updated the test case now.